From 0e18eb85e10c59e090f549312e4100640579fb0c Mon Sep 17 00:00:00 2001
From: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Date: Fri, 23 Mar 2018 17:36:33 -0500
Subject: [PATCH 3542/5242] MLK-17703-9: drm: imx: dcss: align input and
 output pipe gamut and nonlinearity

commit  b262065d8b63edc00afab450d56ca8613b45de8d from
https://source.codeaurora.org/external/imx/linux-imx.git

For better results, output and input pipe gamut and nonlinearity should
match.

Signed-off-by: Laurentiu Palcu <laurentiu.palcu@nxp.com>
Signed-off-by: Meng Li <Meng.Li@windriver.com>
---
 drivers/gpu/drm/imx/dcss/dcss-plane.c |    8 ++++----
 1 file changed, 4 insertions(+), 4 deletions(-)

diff --git a/drivers/gpu/drm/imx/dcss/dcss-plane.c b/drivers/gpu/drm/imx/dcss/dcss-plane.c
index 42da040..aa4aed6 100644
--- a/drivers/gpu/drm/imx/dcss/dcss-plane.c
+++ b/drivers/gpu/drm/imx/dcss/dcss-plane.c
@@ -463,12 +463,12 @@ static void dcss_plane_atomic_update(struct drm_plane *plane,
 
 	ipipe_cfg.pixel_format = pixel_format;
 
-	ipipe_cfg.nl = NL_REC2084;
-	ipipe_cfg.pr = PR_FULL;
-	ipipe_cfg.g = G_REC2020;
-
 	dcss_crtc_get_opipe_cfg(state->crtc, &opipe_cfg);
 
+	ipipe_cfg.nl = opipe_cfg.nl == NL_REC2084 ? NL_REC2084 : NL_REC709;
+	ipipe_cfg.pr = PR_FULL;
+	ipipe_cfg.g = opipe_cfg.g == G_REC2020 ? G_REC2020 : G_REC709;
+
 	dcss_hdr10_setup(dcss_plane->dcss, dcss_plane->ch_num,
 			 &ipipe_cfg, &opipe_cfg);
 
-- 
1.7.9.5

